upload_vehicle_data.py

```diff
@@ -0,0 +1,327 @@
+            if 'receiver' in line:
+                bandwidth_mbps = float(line.split()[-3])
+                return bandwidth_mbps
+    except subprocess.CalledProcessError as e:
+        logging.error(f'iperf3 error: {e}')
+    finally:
+        # Stop iperf3 server on the remote machine
+        stop_server_cmd = (
+            f'sshpass -p {password} ssh {remote_user}@{remote_ip} '
+            f"'pkill iperf3'"
+        )
+        subprocess.run(stop_server_cmd, shell=True)
+        logging.info(f'Stopped iperf3 server on {remote_ip}')
+    return None
+
+
+def main(
+    remote_user,
+    remote_ip,
+    remote_directory,
+    remote_temp_directory,
+    vdi_upload_directory,
+    iperf_server_ip,
+    password,
+):
+    """Automate the upload of rosbags."""
+    # Create the remote temporary directory
+    create_remote_directory(
+        remote_user, remote_ip, remote_temp_directory, password
+    )
+
+    try:
+        # Get the list of rosbags from the remote machine
+        rosbag_list = get_remote_rosbags_list(
+            remote_user, remote_ip, remote_directory, password
+        )
+
+        bandwidth_mbps = measure_bandwidth(remote_ip, remote_user, password)
+        if bandwidth_mbps is None:
+            print('Could not measure bandwidth. Exiting.')
+            return
+
+        total_size_gb = sum(
+            get_remote_file_size(remote_user, remote_ip, rosbag, password)
+            for rosbag in rosbag_list
+        )
+        estimated_time = get_estimated_upload_time(
+            total_size_gb, bandwidth_mbps
+        )
+
+        print(
+            f'Found {len(rosbag_list)} files to upload with total size '
+            f'{total_size_gb:.2f} GB.'
+        )
+        print(f'Measured bandwidth: {bandwidth_mbps:.2f} Mbps')
+        print(f'Estimated upload time: {estimated_time:.2f} hours.')
+        confirm = input('Do you want to proceed to upload? (yes/no): ')
+
+        if confirm.lower() != 'yes':
+            print('Upload aborted.')
+            return
+
+        logging.info(
+            f'Starting upload of {len(rosbag_list)} files with total size '
+            f'{total_size_gb:.2f} GB.'
+        )
+        logging.info(f'Measured bandwidth: {bandwidth_mbps:.2f} Mbps')
+
+        for rosbag in rosbag_list:
+            success = compress_and_transfer_rosbag(
+                remote_user,
+                remote_ip,
+                rosbag,
+                remote_temp_directory,
+                vdi_upload_directory,
+                password,
+            )
+            if not success:
+                logging.error(
+                    f'Failed to process {rosbag}. Continuing with next file.'
+                )
+            else:
+                logging.info(f'Successfully processed {rosbag}.')
+
+    finally:
+        # Delete the remote temporary directory
+        delete_remote_directory(
+            remote_user, remote_ip, remote_temp_directory, password
+        )

```

Comment:
I think here we also need to delete the contents of `remote_directory` for a vehicle's SSD "clean up". Lets implemented and add an option to deactivate this just in case we do not want to delete stuff yet

Reply:
added cleanup option